src/module.c

```diff
@@ -988,18 +989,22 @@ int RMCreateSearchCommand(RedisModuleCtx *ctx, const char *name,
 
   int rc = REDISMODULE_OK;
   char *categories;
-  if (!strcmp(aclCategories, "")) {
-    categories = SEARCH_ACL_CATEGORY;
+  bool is_internal = false;
+
+  if (IS_INTERNAL_COMMAND(name)) {
+    // Internal command, register to the internal ACL category ONLY
+    categories = SEARCH_ACL_INTERNAL_CATEGORY;
```

Comment:
Need further discussion - whether _internal_search category is always separated from other categories or should we also register to additional categories?
* If only _internal_search - it needs to be removed explicitly, (usually admin can define `+@all -@slow` etc. which would not remove _internal_search
* If _internal_search is combined with others, `+@slow` could give users permission to search internal commands without intending to

@MeirShpilraien @LiorKogan